src/components/FormComponents/AssetPicker.js

```diff
@@ -1,87 +1,164 @@
-          issuer: newTypeValue ? "" : value.issuer
-        }))
-      }}
-      className={(isCredit) ? 'picker--spaceBottom' : ''}
-      items={assetButtons}
+  return (
+    <div {...props}>
+      <RadioButtonPicker
+        value={localValue.type}
+        onUpdate={(typeValue) => {
+          const newTypeValue = localValue.type === typeValue ? "" : typeValue;
+          onUpdate(
+            _.assign({}, value, {
+              type: newTypeValue,
+              code: newTypeValue ? "" : localValue.code,
+              issuer: newTypeValue ? "" : localValue.issuer,
+            }),
+          );
+        }}
+        className={isCredit ? "picker--spaceBottom" : ""}
+        items={assetButtons}
       />
-    {codePicker}
-    {codePickerError}
-    {inputPicker}
-  </div>
+      {isCredit && (
+        <>
+          <input
+            type="text"
+            value={value.code}
+            onChange={(event) =>
+              onUpdate(
+                _.assign({}, value, {
+                  code: event.target.value,
+                }),
+              )
+            }
+            placeholder="Asset Code"
+            className="picker picker--textInput"
+          />
+          <PickerError message={codeValidator(value)} />
+          <PubKeyPicker
+            value={value.issuer}
+            onUpdate={(issuer) =>
+              onUpdate(_.assign({}, value, { issuer: issuer }))
+            }
+            placeholder="Issuer Account ID"
+          />
+        </>
+      )}
+    </div>
+  );
 }
 
-function codeValidator(value) {
+function codeValidator(value, ignoreType = false) {
   let minLength, maxLength;
-  if (value.type === 'credit_alphanum4') {
+  if (ignoreType) {
+    minLength = 1;
+    maxLength = 12;
+  } else if (value.type === "credit_alphanum4") {
     minLength = 1;
     maxLength = 4;
-  } else if (value.type === 'credit_alphanum12') {
+  } else if (value.type === "credit_alphanum12") {
     minLength = 5;
     maxLength = 12;
   } else {
     return;
   }
 
-  let code = value.code || '';
+  let code = value.code || "";
 
   if (code && !code.match(/^[a-zA-Z0-9]+$/g)) {
-    return 'Asset code must consist of only letters and numbers.';
+    return "Asset code must consist of only letters and numbers.";
   } else if (code.length < minLength || code.length > maxLength) {
     return `Asset code must be between ${minLength} and ${maxLength} characters long.`;
   }
 }
+
+const stringFormRadioOptions = { native: "Native", issued: "Issued" };
+function AssetStringPicker({ value, onUpdate, ...props }) {
+  const isNative = !value.includes(":");
+
+  return (
+    <div {...props}>
+      <RadioButtonPicker
+        value={isNative ? "native" : "issued"}
+        onUpdate={() => {
+          onUpdate(isNative ? ":" : "native");
+        }}
+        className={!isNative ? "picker--spaceBottom" : ""}
+        items={stringFormRadioOptions}
+      />
+      {!isNative &&
+        (() => {
```

Comment:
Inline [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE). A little awkward, but since the `[code, issuer] = value.split(':')` only works if `!isNative`, this seemed a lot less awkward than the alternative. Handy way to conditionally declare variables within the return.

Reply:
yeah i guess alternatively you could just break this out into a named function and invoke here since it's only being used once, i'm fine w this

Reply:
Yep, it seemed like a worse option to break out into a new component or function. More indirect, by moving single-purpose functionality away from where it's used.